components/mediation/extensions/org.wso2.micro.integrator.transport.handlers/src/main/java/org/wso2/micro/integrator/transport/handlers/requestprocessors/swagger/format/SwaggerYamlProcessor.java

```diff
@@ -51,23 +53,41 @@ public void process(CarbonHttpRequest request, CarbonHttpResponse response,
                         ConfigurationContext configurationContext) throws AxisFault {
 
         API api = getAPIFromSynapseConfig(request);
-        if (api == null) {
+        String responseString = null;
+        if (api != null) {
+            responseString = processAPI(api);
+        } else if (request.getContextPath().contains("/" + SwaggerProcessorConstants.SERVICES_PREFIX)) {
+            responseString = SwaggerUtils.takeDataServiceSwagger(request.getRequestURI(), configurationContext, false);
+        } else {
             handleException(request.getRequestURI());
+        }
+
+        if (responseString != null && !responseString.isEmpty()) {
+            updateResponse(response, responseString, SwaggerConstants.CONTENT_TYPE_YAML);
         } else {
-            //Retrieve from registry
-            String responseString;
+            handleException(request.getRequestURI());
```

Comment:
I suppose we can move this whole code segment to the super class SwaggerGenerator so that we do not have a duplicated code in SwaggerJsonProcessor and SwaggerYamlProcessor?

Reply:
I moved the logic related to API to swagger utils. Now it's consistent. For JSON call getAPISwagger(API,true) for YAML call getAPISwagger(API,false)